### PR TITLE
chore: add type hint for run and async_run

### PR DESCRIPTION
**Pull request by aarnphm**
## What does this PR address?

runner.run() and runner.async_run() before
Screenshot 2022-07-17 at 00 04 12

runner.run() after
Screenshot 2022-07-17 at 00 03 24

This changes only affects TYPE_CHECKING runtime.
## Before submitting:

- [x] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming? Here are [GitHub's
guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) on how to create a pull request.
- [x] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [x] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [ ] Did your changes require updates to the documentation? Have you updated
  those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [ ] Did you write tests to cover your changes?

## Who can help review?

cc @sauyon

### CODE REVIEW

File: bentoml/_internal/runner/runner.py [hunk 1]
```diff
@@ -63,6 +63,10 @@ class Runner:
 
     _runner_handle: RunnerHandle = attr.field(init=False, factory=DummyRunnerHandle)
 
+    if TYPE_CHECKING:
+        run: t.Callable[..., t.Any]
+        async_run: t.Callable[..., t.Awaitable[t.Any]]
```
Comment (ssheng) [inline bentoml/_internal/runner/runner.py:68]: This change looks fine to me but adding the return type as `t.Any` isn't the most useful. Is it possible to derive the return types based on the framework runner implementation?

Comment (aarnphm) [reply bentoml/_internal/runner/runner.py:68]: I think `t.Any` is the best effort here. Another way to do this is to infer the return type from `@svc.api` decorated function.

Comment (aarnphm) [reply bentoml/_internal/runner/runner.py:68]: for the suggestion we can do
```
import inspect
inspect.get_annotations(pipeline.runnable_class.bentoml_runnable_methods__["__call__"].func)
```

Comment (ssheng) [reply bentoml/_internal/runner/runner.py:68]: We will most likely invest more into runner interface declaration when we design the different runtime integration. I'd recommend not to invest too heavily into deriving the runner signature. The best effort here is good for now.

**Review by ssheng — APPROVED** [review]

**Pull request merged**